internal/discovery/hazelcast_cloud_discovery.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discovery
+
+import (
+	"net/http"
+
+	"encoding/json"
+
+	"io/ioutil"
+
+	"strings"
+
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client/core"
+	"github.com/hazelcast/hazelcast-go-client/internal/iputil"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto"
+)
+
+const (
+	cloudURL          = "https://coordinator.hazelcast.cloud/cluster/discovery?token="
+	privateAddressStr = "private-address"
+	publicAddressStr  = "public-address"
+)
+
+type nodeDiscoverer func() (map[string]core.Address, error)
+
+type HazelcastCloud struct {
+	endPointURL               string
+	connectionTimeoutInMillis int64
+	discoverNodes             nodeDiscoverer
+}
+
+func NewHazelcastCloud(cloudToken string, connectionTimeoutInMillis int64) *HazelcastCloud {
+	hzCloud := &HazelcastCloud{}
+	hzCloud.endPointURL = cloudURL + cloudToken
+	hzCloud.connectionTimeoutInMillis = connectionTimeoutInMillis
+	hzCloud.discoverNodes = hzCloud.discoverNodesInternal
+	return hzCloud
+}
+
+func (hzC *HazelcastCloud) discoverNodesInternal() (map[string]core.Address, error) {
+	return hzC.callService()
+}
+
+func (hzC *HazelcastCloud) callService() (map[string]core.Address, error) {
+	url := hzC.endPointURL
+	timeout := time.Duration(hzC.connectionTimeoutInMillis) * time.Millisecond
```

Comment:
Why do we need this time conversion stuff ?
It looks like if we did not convert it in `HazelcastClient.getConnectionTimeoutMillis`(https://github.com/hazelcast/hazelcast-go-client/pull/279/files#diff-c9be6944b26f165a41c612366fa2b4e1R200), then it would not be necessary to convert here, right ?

Reply:
We should keep the type as Duration as much as possible to avoid confusion